modules/nextflow/src/main/groovy/nextflow/extension/CH.groovy

```diff
@@ -30,7 +30,15 @@ class CH {
         return (Session) Global.session
     }
 
-    static private Map<DataflowQueue, DataflowBroadcast> bridges = new HashMap<>(10)
+    static class Topic {
+        String name
+        DataflowBroadcast broadcaster = new DataflowBroadcast()
+        List<DataflowWriteChannel> writers = new ArrayList<>(10)
+    }
+
+    static final private List<Topic> allTopics = new ArrayList<>(10)
```

Comment:
Would it not be safer/clearer to model this as a `Map<String,Topic>` ?

Reply:
yeah